lib/mix/tasks/oban_web.install.ex

```diff
@@ -0,0 +1,140 @@
+defmodule Mix.Tasks.ObanWeb.Install.Docs do
+  @moduledoc false
+
+  def short_doc do
+    "Installs Oban Web into your Phoenix application"
+  end
+
+  def example do
+    "mix oban_web.install"
+  end
+
+  def long_doc do
+    """
+    #{short_doc()}
+
+    This task configures your Phoenix application to use the Oban Web dashboard:
+
+    * Adds the required `Oban.Web.Router` import
+    * Sets up the dashboard route at "/oban" within the :dev_routes conditional
+
+    ## Example
+
+    ```bash
+    #{example()}
+    """
+  end
+end
+
+if Code.ensure_loaded?(Igniter) do
+  defmodule Mix.Tasks.ObanWeb.Install do
+    @shortdoc "#{__MODULE__.Docs.short_doc()}"
+
+    @moduledoc __MODULE__.Docs.long_doc()
+    use Igniter.Mix.Task
+
+    @impl Igniter.Mix.Task
+    def info(_argv, _composing_task) do
+      %Igniter.Mix.Task.Info{
+        group: :oban,
+        installs: [{:oban, "~> 2.0"}],
+        example: __MODULE__.Docs.example()
+      }
+    end
+
+    @impl Igniter.Mix.Task
+    def igniter(igniter) do
+      case Igniter.Libs.Phoenix.select_router(igniter) do
+        {igniter, nil} ->
+          Igniter.add_warning(igniter, """
+          No Phoenix router found, Phoenix Liveview is needed for Oban Web
+          """)
+
+        {igniter, router} ->
+          update_router(igniter, router)
+      end
+    end
+
+    defp update_router(igniter, router) do
+      zipper = &do_update_router(igniter, &1)
+
+      case Igniter.Project.Module.find_and_update_module(igniter, router, zipper) do
+        {:ok, igniter} ->
+          igniter
+
+        {:error, igniter} ->
+          Igniter.add_warning(igniter, """
+          Something went wrong, please check the Oban Web install docs for manual setup instructions
+          """)
+      end
+    end
+
+    defp do_update_router(igniter, zipper) do
+      web_module = Igniter.Libs.Phoenix.web_module(igniter)
+      app_name = Igniter.Project.Application.app_name(igniter)
+
+      with {:ok, zipper} <- add_import(zipper, web_module) do
+        add_route(zipper, app_name)
+      end
+    end
+
+    defp add_import(zipper, web_module) do
+      with {:ok, zipper} <- Igniter.Code.Module.move_to_use(zipper, web_module) do
+        {:ok, Igniter.Code.Common.add_code(zipper, "\nimport Oban.Web.Router")}
+      end
+    end
+
+    defp add_route(zipper, app_name) do
+      with {:dev_routes, {:ok, zipper}} <-
```

Comment:
It doesn't look like the suggestion to compact this was applied. Did the changes break anything?

Reply:
I don't think there was one for this particular piece of code, but I took a stab at it anyway.